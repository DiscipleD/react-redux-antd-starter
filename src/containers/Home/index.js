/**
 * Created by jack on 16-9-6.
 */

import React from 'react';
import {push} from 'react-router-redux';
import {connect} from 'react-redux';
import {createSelector} from 'reselect';

import HeaderNav from 'components/HeaderNav';
import menuActions from 'actions/menu';

@connect(
	state => ({
		...state.menu,
		currentPath: createSelector(
			state => state.menu.list,
			state => state.routing.locationBeforeTransitions.pathname,
			(list = [], pathname) => list.length === 0 ? '' : list.filter(item => item.path === pathname).length > 0 ? pathname : list[0].path
		)(state)
	}), {
		goHome: () => push('/'),
		queryMenuList: menuActions.queryMenuList
	})
class Home extends React.Component {
	componentWillMount() {
		this.props.queryMenuList();
	}

	render() {
		const menuList = this.props.list;
		return (
			<section>
				<HeaderNav
					navList={menuList}
					path={this.props.currentPath}
					onlogoClickFn={this.props.goHome}
				/>
				{this.props.children}
			</section>
		)
	}
}

export default Home;
