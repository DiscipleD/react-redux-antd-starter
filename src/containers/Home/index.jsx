/**
 * Created by jack on 16-9-6.
 */

import React from 'react';

import HeaderNav from 'components/HeaderNav';

export default class Routers extends React.Component {
	render() {
		return (
			<section>
				<HeaderNav path={this.props.location.pathname} />
				{this.props.children}
			</section>
		)
	}
}
