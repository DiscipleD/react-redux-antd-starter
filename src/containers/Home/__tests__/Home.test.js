/**
 * Created by jack on 16-9-20.
 */
import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import middlewares from '../../../redux/middleware';
import Home from '../index';

describe('Home Container', () => {
	const mockStore = configureMockStore(middlewares);

	it('match snapshot', () => {
		// Because of container only includes components
		// so, it's enough to test snapshot only.
		const store = mockStore({
			menu: {},
			routing: {
				locationBeforeTransitions: {
					pathname: '/'
				}
			}
		});
		const appSnapshot = renderer.create(
			<Home store={store} />
		).toJSON();
		expect(appSnapshot).toMatchSnapshot();
	});
});
