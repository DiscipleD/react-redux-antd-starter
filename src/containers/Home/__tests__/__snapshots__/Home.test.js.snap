exports[`Home Container match snapshot 1`] = `
<section>
  <div
    className="header-nav">
    <div
      className="header-nav-wrapper">
      <div
        className="header-nav-logo"
        onClick={[Function anonymous]}>
        Project Logo
      </div>
      <ul
        aria-activedescendant=""
        className="ant-menu ant-menu-horizontal  ant-menu-dark ant-menu-root"
        onKeyDown={[Function bound onKeyDown]}
        role="menu"
        style={
          Object {
            "lineHeight": "64px"
          }
        }
        tabIndex="0" />
    </div>
  </div>
</section>
`;
